src/Aspire.Hosting.MySql/MySqlServerResource.cs

```diff
@@ -14,33 +14,45 @@ public class MySqlServerResource : ContainerResource, IResourceWithConnectionStr
     /// Initializes a new instance of the <see cref="MySqlServerResource"/> class.
     /// </summary>
     /// <param name="name">The name of the resource.</param>
-    /// <param name="password">The MySQL server root password, or <see langword="null"/> to generate a random password.</param>
-    public MySqlServerResource(string name, string? password = null) : base(name)
+    /// <param name="password">A parameter that contains the MySQL server password, or <see langword="null"/> to generate a random password.</param>
+    public MySqlServerResource(string name, ParameterResource? password) : base(name)
     {
         PrimaryEndpoint = new(this, PrimaryEndpointName);
         PasswordInput = new(this, "password");
 
-        Annotations.Add(InputAnnotation.CreateDefaultPasswordInput(password));
+        PasswordParameter = password;
+
+        Annotations.Add(InputAnnotation.CreateDefaultPasswordInput());
     }
 
     /// <summary>
     /// Gets the primary endpoint for the MySQL server.
     /// </summary>
     public EndpointReference PrimaryEndpoint { get; }
 
-    internal InputReference PasswordInput { get; }
+    private InputReference PasswordInput { get; }
+
+    /// <summary>
+    /// Gets the parameter that contains the MySQL server password.
+    /// </summary>
+    public ParameterResource? PasswordParameter { get; }
```

Comment:
Do we want to expose this or the PasswordReference ?

Reply:
I don't want the `InputRefererence` to leak out of the object since the "input" can only be referenced from inside the resource.

Long-term I think we should consider merging Parameters and Inputs. I think the concept of "InputAnnotation" is confusing. Instead we should just have Parameters, and Parameters can have default/generated values.